hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/storage_driver.c

```diff
@@ -463,17 +575,18 @@ static inline void setup8ByteWrite(uint64_t addr, const void *data)
 #endif /* ifdef USING_KSDK2 */
 }
 
+__RAMFUNC
 static inline void setupProgramSection(uint64_t addr, const void *data, size_t cnt)
 {
 #ifdef USING_KSDK2
     static const uintptr_t FlexRAMBase = FSL_FEATURE_FLASH_FLEX_RAM_START_ADDRESS;
-    memcpy((void *)FlexRAMBase, (const uint8_t *)data, cnt);
+    MEMCPY((void *)FlexRAMBase, (const uint8_t *)data, cnt);
```

Comment:
Rohit, what you just wrote in C is what's in that array I posted (apart from the interrupt instructions, which could be added if wanted, and the fact I haven't checked I hand-assembled correctly).

That "just store the code in a non-const array" approach is effectively maximally portable for a Thumb device - totally toolchain independent unless you hit something with a bonkers EABI.

(As a purist, there should probably technically be some sort of barrier instruction before the return instruction, I guess, to prevent prefetch? M3/M4 probably doesn't prefetch returns though.)